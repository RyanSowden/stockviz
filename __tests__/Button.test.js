import React from "react";
import Enzyme from 'enzyme';
import SearchButton from "../components/Button";
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';
import renderer from 'react-test-renderer';
import {shallow, mount} from 'enzyme';




Enzyme.configure({adapter: new Adapter()});

describe("SearchButton", () => {
    it("should render properly", () => {
        const app = renderer.create(<SearchButton />);
        expect(app).toMatchSnapshot();
    });
    it('Should render the stock search form', () =>{
        const wrapper = shallow(<SearchButton />)
        expect(wrapper.find('form[id="form"]').exists()).toBe(true);
    })
    it('the default value should be empty', () => {
		const wrapper = shallow(<SearchButton />)
		expect(wrapper.find('input[id="search"]').prop('value')).toBe();
	})

    it('on change of value in the field, the state of that field in the component should be updated', () => {
		const wrapper = shallow(<SearchButton />)
		expect(wrapper.find('input[id="search"]').simulate('change', {
			target: {
				value: 'test',
			},
        })
        )
    })

    it('Should render the stock search button', () =>{
        const wrapper = shallow(<SearchButton/>)
        expect(wrapper.find('[id="searchbutton"]').exists()).toBe(true);
    })

    it('On submitting, a click handler function should be triggered on click event', () =>{
        const fn = jest.fn();
        const wrapper = shallow(<SearchButton onClick={fn}/>);
        wrapper.find('[id="searchbutton"]').at(0).simulate('click');
        expect(fn).toHaveBeenCalled();
    })
});
