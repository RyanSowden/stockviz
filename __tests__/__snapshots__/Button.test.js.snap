// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchButton should render properly 1`] = `
<div
  className="sc-gsDKAQ fBKVVW"
>
  <form
    id="form"
  >
    <input
      id="search"
      name="search"
      onChange={[Function]}
      placeholder="Stock Ticker"
      style={
        Object {
          "border": "black ridge 1px",
          "padding": "0.6rem",
        }
      }
      type="text"
      vlaue=""
    />
  </form>
  <div
    className="sc-bdvvtL exMHmO"
    id="searchbutton"
    onClick={[Function]}
    style={
      Object {
        "backgroundColor": "#3773f5",
        "color": "black",
      }
    }
  >
    Search
  </div>
  <h1>
    
  </h1>
</div>
`;
